processors/trial_remover/processor.py

```diff
@@ -11,39 +11,129 @@
 # Module API
 
 def process(conf, conn):
-    """Remove trials without records.
+    """Removes trials alongs with their related entities (ex: documents)
+     if they are not related to any other trial or FDA application.
+
+    NOTE: Please run `remove_unknown_documentcloud_docs` processor after this processor.
+    """
+
+    remove_trials_without_records(conf, conn)
+
+
+def remove_trials_without_records(conf, conn):
+    """Remove trials without records"""
+
+    # Prepare
+    query = """
+        SELECT trials.id, trials.identifiers
+        FROM trials LEFT JOIN records ON records.trial_id = trials.id
+        WHERE records.trial_id IS NULL AND trials.source_id != 'pubmed'
     """
 
-    processor = _TrialRemover(conf, conn['database'])
-    processor.remove_trials_without_records()
+    # Execute
+    count = 0
+    for trial in conn['database'].query(query):
+        trial_id = trial['id'].hex
+        remover = _TrialRemover(conf, conn['database'], trial_id)
+        conn['database'].begin()
+        try:
+            remover._delete_related_documents()
+            remover._delete_related_conditions()
+            remover._delete_related_interventions()
+            remover._delete_related_organisations()
+            remover._delete_related_persons()
+            remover._delete_related_locations()
+            remover._delete_related_publications()
+            remover._delete_related_risk_of_biases()
+            remover._delete_trial()
+        except Exception as exception:
+            conn['database'].rollback()
+            logger.exception('Can\'t delete trial: %s and its relations due to error: %s',
+                             trial['identifiers'], repr(exception))
+        else:
+            conn['database'].commit()
+            logger.info('Deleted trial without records: %s and its relations',
+                        trial['identifiers'])
+            count += 1
+        if count and not count % 100:
+            logger.info('Removed %s trials without records', count)
+    logger.info('Removed %s trials without records', count)
```

Comment:
@nightsh See line 58 and 59: If not all the trials (100%) are removed, it will say that only x% of them were.